functional/lenses.md

```diff
@@ -0,0 +1,364 @@
+# Lenses and Prisms
+
+This is a pure functional concept that is not frequently used in Rust.
+Never the less, exploring the concept may be helpful to understand other
+patterns in Rust APIs, such as [visitors](../patterns/behavioural/visitor.md).
+They also have niche use cases.
+
+## Lenses: Uniform Access Across Types
+
+A lens is a concept from functional languages that allows accessing parts of a
+data type in an abstract, unified way.[^1]
```

Comment:
"parts of a data type" sounds kind of strange, but I don't currently have a better idea how to express this either.